### ISSUE TITLE
Pushing an incremented package version

### ISSUE THREAD

**Issue by jhawk-kar** [azlamsalam/sfpowerscripts#11]
I have a question about push changes to the `sfdx-project.json` file from ADO after it has been committed locally.  I have tried running a script after the create version task is complete, but whenever I run git status, it doesn't see any changes being reporting.  Can you give me a quick rundown of how you are interacting with the detached head and how I could push the commit to our repository?

**Comment by azlam-abdulsalam** [comment]
@jhawk-kar  I am not clear on the question, Is it that you need to do something after pushing an incremented version?

You could do something similar to the script here.

https://github.com/azlamsalam/sfpowerscripts/blob/master/BuildTasks/IncrementProjectBuildNumberTask/IncrementProjectBuildNumber.ts

**Comment by jhawk-kar** [comment]
Sorry, I'll try to give some more detail.  We are using that task you referenced and are doing the commit to the `sfdx-project.json` file to increment the build number.  We then use the create package version in salesforce task.  After that, I want to push the commit to our repository, so that we are maintaining the accurate build number in the project.  I'm trying to do that through a bash task.  When I run git status in the script, it doesn't see that any changes were made through the commit step, but the logs do show that the commit was made.  I'm just unclear the right way to push the committed changes to our repository and I think it is the detached HEAD that is confusing me.  If you have an idea of how to accomplish the push step, I'd appreciate the advise.

**Comment by azlam-abdulsalam** [comment]
@jhawk-kar  This is what you would do if your version control is github,
git push https://$(github_pat)@github.com/<org>/<repo>.git HEAD:<branch>

It would be pretty similar for other version controls, minor differences in the PAT section.  I could add this as a separate task later.

I would like to touch base with you to gain further feedback. would you be able to join our slack channel.. https://join.slack.com/t/sfpower/shared_invite/enQtNzcxMjE1MTg0ODIzLWFmNDRkY2NkOGE0NDkxMWVmZmQ1ODJhMGVhZTkzYjYzOWMyNWZiY2Y1YTlhZmU2NDAyMDVlODMyMjU1NWZmZjA

**Comment by jhawk-kar** [comment]
Thanks, @azlam-abdulsalam.  I think I figured out where I was going wrong.  I was trying to push to the HEAD of the PR commit instead of the PR source branch.  The following worked for me:
`git push origin HEAD:<pr-source-branch>`
I passed in the `System.PullRequest.SourceBranch` from ADO to my script and it work exactly as it was supposed to.  

Thank you for the slack invite.  Feel free to reach out if needed.
